packages/next/src/client/components/client-page.tsx

```diff
@@ -1,30 +1,107 @@
 'use client'
 
+import type { ParsedUrlQuery } from 'querystring'
+import { use } from 'react'
+import { InvariantError } from '../../shared/lib/invariant-error'
+
+import type { Params } from '../../server/request/params'
+
 export function ClientPageRoot({
-  Component,
-  props,
+  C /* Component */,
+  sp /* searchParams */,
+  p /* params */,
+  up /* underlying params */,
+  fp /* fallback param names */,
 }: {
-  Component: React.ComponentType<any>
-  props: { [props: string]: any }
+  C /* Component */ : React.ComponentType<any>
+  sp /* searchParams */ : Promise<ParsedUrlQuery>
+  p /* params */ : Promise<Params>
+  up? /* underlying params */ : Params
+  fp? /* fallback param names */ : Set<string>
```

Comment:
This only is rendered once per segment, the prop names are descriptive and should be preserved.

Worst case, the type should be extracted and deeply commented using JSDoc style comments.

Reply:
Yeah'll just change it back